### PR TITLE
Integration Tests for Throttle Policy Reset Support Feature

### PR DESCRIPTION
**Pull request by Chanuka-ChandraYapa**
Related Issue: https://github.com/wso2/api-manager/issues/2455

Related PRs
- carbon PR: https://github.com/wso2/carbon-apimgt/pull/12443
- UI PR: https://github.com/wso2/apim-apps/pull/700

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+                restAPIStore.resetApplicationThrottlePolicy(
+                requestCountApplicationDTO.getApplicationId(), userId);
+        Assert.assertEquals(resetResponse.getStatusCode(), 200, "reset Application policy is not successful");
+        Thread.sleep(5000);
+
+        // invoke the api and check the throttling again to verify reset is happened
+        checkThrottling(apiInvocationUrl, requestHeaders, 5);
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test reset application throttling policy with request count limit")
```
Comment (rmsamitha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java:213]: Description incorrect

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+        // Create an Application for bandwidth limit
+        bandwidthApplicationDTO = restAPIStore.addApplication(applicationName2, policyName2, "",
+                "this-is-test-application-for-bandwidth-limit");
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test reset application throttling policy with request count limit")
+    public void testResetPolicyWithRequestCountLimit() throws Exception {
+        //subscribe to API
+        SubscriptionDTO subscriptionDTO = restAPIStore.subscribeToAPI(apiId,
```
Comment (rmsamitha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java:174]: Move subscription code to setEnvironment method

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                "3600", null, ApplicationKeyGenerateRequestDTO.KeyTypeEnum.PRODUCTION,
+                null, grantTypes);
+        String accessToken = applicationKeyDTO.getToken().getAccessToken();
+
+        Map<String, String> requestHeaders = new HashMap<String, String>();
+        requestHeaders.put("Authorization", "Bearer " + accessToken);
+        requestHeaders.put("accept", "text/xml");
+        requestHeaders.put("content-type", "application/json");
+        String apiInvocationUrl = getAPIInvocationURLHttps(APIContext + "/" + APIVersion + "/");
+
```
Comment (rmsamitha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java:193]: Code untill here should go to setEnvironment method

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
```
Comment (rmsamitha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java:235]: Code untill here should go to setEnvironment method

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+        grantTypes.add(APIMIntegrationConstants.GRANT_TYPE.CLIENT_CREDENTIAL);
+        ApplicationKeyDTO applicationKeyDTO = restAPIStore.generateKeys(requestCountApplicationDTO.getApplicationId(),
+                "3600", null, ApplicationKeyGenerateRequestDTO.KeyTypeEnum.PRODUCTION,
+                null, grantTypes);
+        String accessToken = applicationKeyDTO.getToken().getAccessToken();
+
+        Map<String, String> requestHeaders = new HashMap<String, String>();
+        requestHeaders.put("Authorization", "Bearer " + accessToken);
+        requestHeaders.put("accept", "text/xml");
+        requestHeaders.put("content-type", "application/json");
```
Comment (rmsamitha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java:191]: There are already defined content type constants for the values of these content types

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java (hunk 1 above)
Comment (Chanuka-ChandraYapa) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java]: Fixed with https://github.com/wso2/product-apim/pull/13493/commits/ea0b01ff6b639e1245e49754bb332e3fcf359753

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java (hunk 2 above)
Comment (Chanuka-ChandraYapa) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java]: Fixed with https://github.com/wso2/product-apim/pull/13493/commits/ea0b01ff6b639e1245e49754bb332e3fcf359753

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java (hunk 3 above)
Comment (Chanuka-ChandraYapa) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java:200]: Fixed with https://github.com/wso2/product-apim/pull/13493/commits/ea0b01ff6b639e1245e49754bb332e3fcf359753

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java (hunk 4 above)
Comment (Chanuka-ChandraYapa) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java]: Fixed with https://github.com/wso2/product-apim/pull/13493/commits/ea0b01ff6b639e1245e49754bb332e3fcf359753

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java (hunk 5 above)
Comment (Chanuka-ChandraYapa) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java]: Fixed with https://github.com/wso2/product-apim/pull/13493/commits/ea0b01ff6b639e1245e49754bb332e3fcf359753

**Review by rmsamitha — APPROVED** [review]

**Pull request merged**